src/main/java/teammates/ui/webapi/request/FeedbackResponseSubmissionRequest.java

```diff
@@ -0,0 +1,64 @@
+package teammates.ui.webapi.request;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import teammates.common.datatransfer.questions.FeedbackQuestionType;
+import teammates.common.datatransfer.questions.FeedbackResponseDetails;
+import teammates.common.util.JsonUtils;
+
+/**
+ * The basic request of modifying a feedback response.
+ */
+public class FeedbackResponseSubmissionRequest extends BasicRequest {
+
+    private FeedbackQuestionType questionType;
+    private Map<String, Map<String, Object>> responseDetailsPerRecipient = new HashMap<>();
```

Comment:
I don't recommend to use map as API request. It is too abstract and there is no way to tell the structure immediately even looking at the class itself.

Instead, I think it should accept a list of `FeedbackResponseRequest` where there are two field `recipientIdentifier ` and `responseDetails`.